src/main/scala/org/scalameter/Parameters.scala

```diff
@@ -1,20 +1,31 @@
 package org.scalameter
 
+import scala.language.existentials
 import scala.collection.{Iterable, immutable}
+import Parameters.P
+
 
 @SerialVersionUID(4203959258570851398L)
-case class Parameters(axisData: immutable.ListMap[String, Any]) {
+case class Parameters(axisData: immutable.ListMap[Parameter[P], P]) {
   def ++(that: Parameters) = Parameters(this.axisData ++ that.axisData)
-  def apply[T](key: String) = axisData.apply(key).asInstanceOf[T]
-  def map(f: ((String, Any)) => (String, Any)) = Parameters(axisData.map(f))
+  def apply[T: Manifest](key: String) = axisData.apply(Parameter[T](key)).asInstanceOf[T]
```

Comment:
I suggest you keep it simple, and make `Parameters` a map from `String` to `(Any, Pickler[Any])`.